test/types/__init__.py

```diff
@@ -0,0 +1,13 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests by type."""
```

Comment:
Why do we need the `types` directory? Could we just have `test/unit`,`test/integration`, `test/e2e` and `test/utils` to keep the directory structure shallow?

Reply:
Good question. It's a tradeoff. 

Used what you proposed in a first shot, too, and then decided to introduce an additional `types` directory because I didn't like that root folders for different _types_ were at the same level with other common test folders (e.g. `mock`, `utils`) 

```
|-test
|--e2e
|--integration
|--mock
|--unit
|--utils
```

What do you think? Other proposals that would combine "the best of both worlds" ;)?

Reply:
Thought a bit more about how to combine both ideas and came up with a solution via https://github.com/Qiskit/qiskit-ibm-runtime/pull/139/commits/4e489194c793b796600ea65c35f5445fda10468b that leads to below shallow structure:

<img width="215" alt="image" src="https://user-images.githubusercontent.com/16404496/153184935-6d7104ca-6aad-4a94-ace6-adb99d05d265.png">